src/components/data-table/column-sorting/column-sort.service.ts

```diff
@@ -0,0 +1,52 @@
+import {Injectable} from '@angular/core';
+import {BehaviorSubject} from 'rxjs/BehaviorSubject';
+
+/** Specifies sorting direction */
+export enum SortDirection {
+  ASCEND, DESCEND
+}
+
+/** Sorting model. */
+export interface ColumnSortingModel {
+  column: string;
+  direction: SortDirection;
+}
+
+
+/**
+ * Service handling column sort change events as an observable stream.
+ */
+@Injectable()
+export class MdDataColumnSortingService {
+  public sortingColumn$: BehaviorSubject<ColumnSortingModel> = new BehaviorSubject(
+      { column: 'NONE', direction: SortDirection.ASCEND });
+
+  /**
+   * Programatically set the column / direction of sorting
+   * @param sorting column & direction to sort by.
+   */
+  public setSorting(sorting: ColumnSortingModel) {
+    this.sortingColumn$.next(sorting);
```

Comment:
It's conventional way indicate a var is an observable... but I'll probably have to take it out, because it's not really a convention followed in the code-base so far.